### DISCUSSION TITLE
Price to tick calculation

### DISCUSSION THREAD

**Discussion by adradr** [Jeiwan/uniswapv3-book#58] [Milestone 1. First Swap] [answered]
I am probably missing some point here, but [after reading your article](https://uniswapv3book.com/docs/milestone_1/calculating-liquidity/), still don't understand.

I copied a random position from the USDC/ETH 0.3% Ethereum mainnet pool. By simply running your `price_to_tick` function over this `1900` price it yields a totally different tick price. Can you please elaborate what am I missing here? 

Thanks,
Adrian

EDIT: 0.05% -> 0.3%

https://github.com/Jeiwan/uniswapv3-code/blob/0924bbd239cb153a9975a83c5d376f42e839f457/unimath.py#L10-L11

![Screenshot 2023-01-30 at 19 18 17](https://user-images.githubusercontent.com/9167422/215561805-793f2397-4b78-4a44-84ae-0d9d3a4a45df.png)

<img width="846" alt="Screenshot 2023-01-30 at 19 17 30" src="https://user-images.githubusercontent.com/9167422/215561645-fdaa9474-92e9-494e-a153-fd82fa845e1d.png">

**Comment by Jeiwan** [comment, marked as answer]
GitHub wiped out my answer... Well, in short, you need to flip the price so it's the price of USDC in WETH (the price in a pool is the price of token 0 in terms of token 1):
```python
math.floor(math.log(int(1e18 // 1900.79e6), 1.0001))
# 200820
```
